library/objective-c/EnvoyConfiguration.m

```diff
@@ -198,11 +200,20 @@ - (nullable NSString *)resolveTemplate:(NSString *)templateYAML {
                             self.appVersion, self.appId];
   [definitions appendFormat:@"- &virtual_clusters %@\n", self.virtualClusters];
 
+  [definitions
+      appendFormat:@"- &stats_flush_interval %lus\n", (unsigned long)self.statsFlushSeconds];
+
+  NSMutableArray *stat_sinks_config = [self.statsSinks mutableCopy];
+
   if (self.grpcStatsDomain != nil) {
     [definitions appendFormat:@"- &stats_domain %@\n", self.grpcStatsDomain];
-    [definitions
-        appendFormat:@"- &stats_flush_interval %lus\n", (unsigned long)self.statsFlushSeconds];
-    [definitions appendString:@"- &stats_sinks [ *base_metrics_service ]\n"];
+    [stat_sinks_config addObject:@"*base_metrics_serivce"];
+  }
+
+  if ([stat_sinks_config count] > 0) {
```

Comment:
```suggestion
  if (stat_sinks_config.count > 0) {
```
Missed previously - dot accessor is a preferred one for things that are implemented as properties (even though your implementation works just fine).